src/electron/routing_service.ts

```diff
@@ -279,16 +291,43 @@ async function installLinuxRoutingServices(): Promise<void> {
   const srcFolderPath = path.join(getAppPath(), OUTLINE_PROXY_CONTROLLER_PATH);
 
   console.log(`copying service installation files to ${tmp}`);
-  for (const [filename, executable] of LINUX_SERVICE_FILE_NAMES) {
-    const dest = path.join(tmp, filename);
-    await fsextra.copy(path.join(srcFolderPath, filename), dest, {overwrite: true});
-    if (executable) {
+  for (const descriptor of LINUX_SERVICE_FILE_DESCRIPTORS) {
+    const src = path.join(srcFolderPath, descriptor.filename);
+
+    const srcContent = await fsextra.readFile(src);
+    descriptor.sha256 = createHash('sha256')
+      .update(srcContent)
+      .digest('hex');
+
+    const dest = path.join(tmp, descriptor.filename);
+    await fsextra.copy(src, dest, {overwrite: true});
+
```

Comment:
You could call `chattr -i` and check the checksum here, as the user.
Also, no need to use the `shasum`, since you can calculate it easily in node (perhaps move it to a function).

The only thing you need root for is to actually run the script after validation and to remove the immutable flag in the end.

Reply:
`chattr -i` means "remove the immutable flag", so it makes the files mutable. So we should use `chattr +i` to make sure the files are not writable. In addition, both `chattr +i` and `chattr -i` are required to run under root.